general/src/store/trimming.rs

```diff
@@ -359,15 +359,17 @@ fn slim_frame_data(frames: &mut Array<Frame>, frame_allowance: usize) {
 
     let app_count = app_frame_indices.len();
     let system_allowance_half = frame_allowance.saturating_sub(app_count) / 2;
-    let system_frames_to_remove = &system_frame_indices
-        [system_allowance_half..system_frame_indices.len() - system_allowance_half];
+    let system_frames_to_remove = system_frame_indices
+        .get(system_allowance_half..system_frame_indices.len() - system_allowance_half)
```

Comment:
This will return `None` if the index is out of bounds, instead of capping. Is that what you want here?

Reply:
I wanted to prevent the situation where the end index is smaller than the start index. I wasn't able to repro the behavior.

If the index is out of bounds I believe that the allowance is larger than the array (I see no other way). That's also fine, in that case nothing has to be stripped.

Generally I would just like to avoid panicking, even if the result is wrong then.

Reply:
To leave a comment here: I think it's absolutely find to generally not panic. This pattern we use elsewhere as well and the python version also just produces an empty slice here.

Reply:
Ah I see, Python really does produce an empty slice always. I was under the assumption that some things do trigger IndexError.